### DISCUSSION TITLE
2024-07-18 목요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#54] [기록보드]
## 2024-07-18 목요일 기록보드

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 기술면접 / 답안작성 | |
| 10:00 ~ 10:00 | 하루 일과 준비 | |
| 10:00 ~ 13:00 | 디자이너 미팅(와이어프레임 작성) | <img src="https://github.com/user-attachments/assets/90d42196-1329-405b-98b5-8b46552df58a" weight="150" height="150"/> <img src="https://github.com/user-attachments/assets/24012912-4606-405c-bce1-ddf58030ab32" weight="150" height="150"/> |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 와이어 프레임 작성 및 개인별 의사코드 작성 | <img src="https://github.com/user-attachments/assets/b047f4e7-602d-4743-98ce-4a68878f05f2" weight="300" height="300"/> |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 개인별 의사코드 공유, 디자이너 미팅 | <img src="https://github.com/user-attachments/assets/f7fdce6f-a4db-45b4-8700-e52d87f4b415" weight="300" height="300"/> |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

## 2024-07-19 금요일 예정사항

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 개발자 회의 (의사코드 공유) | |
| 10:00 ~ 13:00 | 오전 디자이너 미팅 (와이어프레임 확정) | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 13:00 ~ 16:00 | 오후 디자이너 미팅 (와이어프레임 확정) | |
| 16:00 ~ 18:00 | 의사코드 공유 + DB 테이블 설계 | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 의사코드 공유 + DB 테이블 설계 + 페이지 및 컴포넌트 분배 확정 | |
| 20:00 ~ 21:00 | 회고 | |

### 오늘 한 일

- 본인인증 외부 API 응답값 확인 및 테스트
```
// 사진
<img src="이미지 주소" weight="300" height="300"/>
```

### 내일 계획

- supabase auth 비밀번호 찾기 기능 연결

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 와이어 프레임 작성
- 디자이너 미팅
- 디자인 멘토 피드백 청취
- 개인 의사 코드 작성

### 내일 계획

- 와이어 프레임 작성
- API 명세서 작성

### 개인 의사 코드

#### 로그인/회원가입

- 튜토리얼 완료 후
- 유저가 (1) 회원가입을 누르거나 (2) 비로그인 상태로 글쓰기, 채팅버튼, DM 버튼, 팔로우 버튼을 눌렀을 때 login 페이지로 리다이렉션 -> 이곳에서 회원가입 버튼 렌더링 -> 회원가입에서는 4 + 1가지 옵션 제공
  - 1) `supabase.auth.signUp` 메서드로 email, password 회원 가입 [docs](https://supabase.com/docs/reference/javascript/auth-signup)
  - 2) `supabase.auth.signInwithAuth` 메서드로 kakao proivder 로그인 [docs](https://supabase.com/docs/guides/auth/social-login/auth-kakao)
  - 3) `suabase.auth.signInwithAuth` 메서드로 google proivder 로그인 [docs](https://supabase.com/docs/guides/auth/social-login/auth-google)
  - 4) `supabase.suth.signInwithAuth` 메서드로 apple provider 로그인 [docs](https://supabase.com/docs/guides/auth/social-login/auth-apple)
  - 5) option naver 소셜 로그인 [docs](https://developers.naver.com/docs/login/overview/overview.md)

#### 비밀번호 찾기 (password reset request)

- 로그인 페이지에서 `계정을 분실하셨습니까?` 링크 클릭 시 비밀번호 찾기 페이지로 리다이렉트 [docs](https://supabase.com/docs/reference/javascript/auth-resetpasswordforemail)

#### 로그아웃 [docs](https://supabase.com/docs/reference/javascript/auth-signout)

- 상단 헤더에서 `로그아웃` 링크 클릭 -> '정말 로그아웃 하시겠습니까?' prompt -> 로그아웃 처리 -> home으로 리다이렉트

#### 온보딩

- 사용자 회원 가입 후 `isFinishedOnboarding` 컬럼 false면 onboarding page로 리다이렉트 (회원가입 후 buddies 테이블에 user row 생성 시 해당 컬럼 default value 'false' -> onboarding 완료 후 '제출하기' 버튼 클릭 시 true로 변경 -> 이 컬럼이 true이거나 undefined인 경우(비로그인) url path로 접근 시 middleware에서 home으로 리다이렉트
- 온보딩 페이지는 총 4 페이지 (seq 전달 방식).
  - 1) '버디님 안녕하세요! 몇 가지 추가 정보를 입력하시면 버디님께 꼭 맞는 여정과 버디를 추천해드릴게요!'
    - '다음에' 클릭 -> home 리다이렉트
    - '입력하기' 클릭 -> 다음 seq로 이동(온보딩 진행)
  - 2) '버디님의 추가 정보를 적어주세요.'
    - null 비허용, 입력필드 유효성 검사 및 alert 경고
    - 프로필사진(webp 변환, optimistic, 파일명은 user.id로 덮어쓰기 방식), 닉네임, 거주지역, MBTI
    - NICE,PASS, etc... 인증 가능한가? => 실명, 전화번호, 생년월일, 성별
  - 3) 선호하는 여정 테마 -> chips 제공 -> 선택 시 preferred_trip_themes 컬럼에 jsonb 형태로 저장, 선호하는 버디 -> chips 제공 -> 선택 시 preferred_trip_characters 컬럼에 jsonb 형태로 저장

#### 여정(파티) 만들기

#### 여정(파티) 수정하기

#### 버디즈 랭킹 리스트

#### 마이페이지(프로필)

**Comment by eunohhh** [comment]
## 개인 기록보드서식 (오은)

### 오늘 한 일
- 나온 와이어프레임을 바탕으로 전체 점검
- webRTC 테스트 계속
- 음성통신 / 스토리 페이지 와이어프레임 초안 작성
 
- 음성채팅 의사코드 작성해보기
  - 1. /chat/[id] 에서 음성채팅을 누르면
  - 2. 라우트가 바뀌지 않고, 전체화면 모달이 뜸
  - 3. 약간의 로딩??
  - 4. 최대 4명이므로 사용자의 프로필을 4개까지 보여줌
  - 5. 4번의 방법으로는 /chat 에서 동시접속자정보를 props 로 받음?
  - 6. simple-peer 를 사용하여 
  - 7. `<audio />` 태그를 4개 까지 준비해놓고
  - 8. navigator.mediaDevices 지원여부를 확인 < 어디서?
  - 9. 사용자 동작이 모바일에서는 필수이므로 클릭이벤트핸들러에서
  - 10. 지원여부가 확인되면 getUserMedia()
  - 11. (peer 1개 자기 목소리 들리는 것 까지는 테스트 완료)
  - 12. 4개까지 된다고하니... new Peer 4개 해서 테스트 해볼것
  - 13. peer.on("error", (err)=>{})
  - 14. peer.on(“signal”, (data)=>{})
  - 15. peer.on(“stream”, (stream) => {})
  - 16. 위 세가지 메서드로 통제 
  - 17. 각 `<audio />` 태그에 ref 를 달아서 15번 stream 시
  - 18. srcObject 혹은 src 가 HTMLAudioElement 에 있으면
  - 19. srcObject의 경우 바로 .current = stream
  - 20. src 의 경우 URL.createObjectURL(new Blob([ stream ], { type : audio/ogg }))
  
- 스토리 의사코드 작성해보기
  - 1. /[id]에서, 만일 [id] === 자신.id 면
  - 2. 스토리 추가 버튼이 있고, 해당 버튼을 누르면
  - 3. /addStory 로 이동하거나, 전체화면 모달을 띄우거나(이동이 나아보임)
  - 4. 사진첩에서 선택의경우 파일 인풋으로 사진 받고 받아지면 시퀀스 이동
  - 5. (모바일) 카메라버튼 선택의 경우 navigator.mediaDevices.getUserMedia(options)
  - 6. options = { video:  {   facingMode: { exact: "environment" }   } }
  - 7. (이게 될까? 된다면 다음으로)
  - 8. 7번이 성공한다면 촬영된 사진을 가지고 다음 시퀀스로 이동
  - 9. 배경에는 촬영하거나 선택된 사진이 있고
  - 10. 상단에 텍스트 / 필터 / 스티커???(미정) 이런 버튼들이 있음
  - 11. 필터는 테일윈드에 아주 잘되어 있으므로 누를때마다 순차적으로 사진에 필터 적용하면되고
  - 12. 텍스트는 오버레이 `<div>`를 하나 만들어놓고
  - 13. 기본 중앙에서 input type=”text” 해서 인풋 받고
  - 14. absolute fixed 주고 드래그 되도록... (이것은 라이브러리 찾아볼것)
  - 15. 아무튼 드래그가 되어서, onMouseLeave?? 시에 pageX, pageY??? 이게 맞나
  - 16. 암튼 그 값을 찾아서 overlay 칼럼에 jsonb 로 저장... { overlay : [ { text : “값", position: [ x, y] }, {}, ... ]}
  - 17. 여기까지 되면 성공
  - 18. (스토리보여주는 페이지에서는 저것을 이용해 역시 overlay <div> 내에 map 돌려서 보여주면 될...까...?)  

### 내일 계획
- Supabase DB 세팅 계속
- 디자이너 미팅
- 실제 코드를 짜보기(시간되면)

**Comment by Kyara0324** [comment]
### 오늘 한 일

- 와이어 프레임 작성
- 디자이너 미팅
- 디자인 멘토 피드백 청취


### 내일 할 일

- 와이어 프레임 작성
- API 명세서 작성

**Comment by slsl2** [comment]
## 오늘 한 일 (다같이)

- 와이어프레임 피드백
- 세부 기능, 로직, 플로우 정리
- 디자이너 미팅

## 할 일 (개인)

- 담당 기능 의사 코드 작성
- 담당 데이터베이스 구성 수정
- 디자인 관련 분위기 등 생각해보기

**Comment by hyunjeongkwak** [comment]
### 오늘 한 일

- 디자이너님과 미팅
- 와이어 프레임 수정 및 세부 기능 정리
- 개인별 의사코드 작성해보기
  - 검색
  - 1. 키워드 검색 : 키워드 입력 - 키워드 제공되었으면 필터링된 검색 결과 페이지로 이동(자동 완성, 오타 수정)
    - 1. 가이드 문서 배포 : Jenkins를 통해 guide문서를 guide 서버에 배포
    - 2. 가이드 문서 parsing : Docsearch를 이용하여 가이드 문서를 parsing
    - 3. algolia service로 data 인덱싱: parsing 한 데이터는 Docsearch에 의해 자동으로 algolia service에 인덱싱된다고 함
    - 4. 가이드 페이지에서 검색 : 검색 시 docsearch.js를 활용하여 algolia api를 호출
    - 5. api호출에 대한 response: 검색 결과를 받아 화면에 노출

  - 2. 지역 및 국가 검색 : 위치 검색 모달 -  국내/해외 선택 토글 - 하위 요소(지역/대륙 및 국가, 지역) 선택 후 검색 결과 페이지로 이동
  - 3. 날짜 및 기간 검색 : 날짜 선택 모달(달력 라이브러리) - 날짜 선택 후 검색 결과 페이지로 이동
  - 4. 여정 테마 또는 버디즈 성향이 제공된다면
    - 여정 테마 있다면, 여정 테이블에서 선택한 여정 유형 일치 여부 확인, 일치하는 여정 유형 개수 카운트
    - 버디즈 성향이 있다면, 여정 테이블에서 선호 버디즈 유형 일치 여부 확인, 일치하는 버디즈 유형 개수 카운트
  - 5. 검색결과페이지 정렬
    - 여정 검색 결과 : 일치하는 칩스 카운트가 높을수록 위에 뜸
    - 성향 맞는 버디즈 추천 : 온보딩에 작성된 본인의 성향과 칩스 카운트 높은 순 몇 명만 추천

  - 검색 결과 페이지

### 내일 할 일
- 데이터베이스 구성 수정
- 디자인 레퍼런스 찾아서 공유하기
- 코드 작성?
